arrow-libs/fx/arrow-fx-coroutines/src/commonTest/kotlin/arrow/fx/coroutines/BracketCaseTest.kt

```diff
@@ -206,58 +245,66 @@ class BracketCaseTest {
     }
   }
 
-  operator fun Throwable.plus(other: Throwable): Throwable =
-    apply { addSuppressed(other) }
+  private infix fun Throwable.shouldHaveSuppressed(exception: Throwable) =
+    suppressedExceptions shouldBe listOf(exception)
 
   @Test
   fun bracketCaseMustComposeImmediateUseAndImmediateReleaseError() = runTest {
-    checkAll(10, Arb.int(), Arb.throwable(), Arb.throwable()) { n, e, e2 ->
+    checkAll(10, Arb.int(), Arb.throwableConstructor(), Arb.throwable()) { n, eConstructor, e2 ->
+      val e = eConstructor()
       Either.catch {
         bracketCase<Int, Unit>(
           acquire = { n },
           use = { throw e },
           release = { _, _ -> throw e2 }
         )
-      } shouldBe Either.Left(e + e2)
```

Comment:
I am very surprised by this change of behavior. Could you explain it?

Reply:
Yes sorry I noticed this while refactoring, should've probably gone in a different PR.
This is not a change of behaviour, it's just fixing a misleading test. The previous test didn't test what it claimed to.
Basically, the `plus` method did not do what it seems to do. Let me demonstrate this with (mutable) lists:
```
val e2 = ...
val e = mutableListOf<Throwable>()
e.add(e2)

e shouldBe (e.apply { add(e2) })
```
As you can see, this isn't actually testing that e2 is contained in e, in fact, this would also pass the same test:
```
val e2 = ...
val e = mutableListOf<Throwable>()

e shouldBe (e.apply { add(e2) })
```
In other words, the previous test would pass if you do:
```
Either.catch {
  bracketCase<Int, Unit>(
    acquire = { n },
    use = { throw e },
    release = { _, _ -> }
  )
} shouldBe Either.Left(e + e2)
```
because it's not actually testing inclusion because that `+` is misleading.

Reply:
Also, I had to use a constructor for throwables for the same mutability reason: the throwable might get reused across iterations of `checkAll`, and hence we actually ended up with multiple suppressed exceptions in `e`.

Reply:
I can revert this if you want and add it to a different PR because this has nothing to do with the behaviour changes in this PR

Reply:
Ah, I see, this is because of the `operator plus` on `Throwable`! Now I understand, I was getting confused by how `plus` on lists actually does what you expect (create a new copy), but this is not the case for the implementation above!